templates/default.nginx_http.conf.template

```diff
@@ -25,6 +25,20 @@ add_header X-Frame-Options DENY;
 # to make sure they are authentic. This will become obsolete in June 2021.
 add_header Expect-CT 'enforce, max-age=604800';
 
+# Enable cross-site scripting filter (XSS)
+# (handled by the browser; usually enabled by default,
+# this header will re-enable the filter if disabled by user.)
+add_header X-XSS-Protection "1; mode=block";
+
+# Explicitly list domains allowed to serve content for this site
+add_header Content-Security-Policy-Report-Only "default-src 'self'; object-src 'none'; base-uri 'self'; manifest-src 'self'; media-src 'self'; worker-src 'none';";
+add_header Content-Security-Policy-Report-Only "script-src 'report-sample' 'self' https://discuss.dockstore.org https://gui.dockstore.org https://platform.twitter.com https://www.google-analytics.com https://www.googletagmanager.com;";
+add_header Content-Security-Policy-Report-Only "style-src 'report-sample' 'self' https://cdnjs.cloudflare.com https://fonts.googleapis.com https://gui.dockstore.org;";
+add_header Content-Security-Policy-Report-Only "connect-src 'self' https://s3.amazonaws.com https://api.github.com https://view.commonwl.org;";
+add_header Content-Security-Policy-Report-Only "font-src 'self' https://fonts.gstatic.com https://gui.dockstore.org;";
+add_header Content-Security-Policy-Report-Only "frame-src 'self' https://discuss.dockstore.org;";
+add_header Content-Security-Policy-Report-Only "img-src 'self' data: https://*.githubusercontent.com https://gui.dockstore.org https://i.imgur.com https://img.shields.io https://quay.io https://via.placeholder.com https://www.googletagmanager.com https://www.gravatar.com;";
```

Comment:
Is it intended to support rendering of arbitrary images in Dockstore readme content? Github is able to support this by rendering caching offsite images to their domain. You use a shields.io link but they host it from `camo`

https://camo.githubusercontent.com/65f5a279d0cb5de30f8d26a978ad27d9290d5e2a/68747470733a2f2f7a656e6f646f2e6f72672f62616467652f444f492f31302e353238312f7a656e6f646f2e333632383739312e737667

This might be out of the scope of this ticket but this line makes me curious of what we intend for the readme display feature. Adding imgur to our security policy seems like it would allow more than we intended.

Reply:
@david4096 there is no way to allow arbitrary content using the CSP, so no, as long as there is a CSP we will only display images whose top level domain is explicitly allowed. We would need caching of offsite images to get around that. (To better understand the reason for this, images are usually transmitted/stored in compressed format, and some maliciously-crafted images can execute arbitrary code when decompressed, so images should be treated as having an equal potential for bad things as Javascript.)

I agree that allowing images from imgur (and a potentially very long list of other custom domains) is not a great approach, so we can add those to the CSP header until we implement caching (or find some other fix).

w.r.t. the shields.io link, some of the badges on the dockstore pages for different projects were images hosted at shields.io, so that's why they show up in the CSP. I think Github does its own image caching, which is why the source appears to be camo.githubusercontent.com. (Also, we set a wildcard to allow `*.githubusercontent.com` so the camo link should be allowed.)

Reply:
@david4096 You raised a great point about raw.githubusercontent.com, and the fact that allowing `*.githubusercontent.com` would potentially allow some arbitrary user content, so I replaced the wildcard with `{avatars0,avatars1,avatars2,avatars3,camo}.githubusercontent.com`